pdns/dnsdist-lua-actions.cc

```diff
@@ -2416,8 +2422,8 @@ void setupLuaActions(LuaContext& luaCtx)
     });
 #endif /* DISABLE_PROTOBUF */
 
-  luaCtx.writeFunction("TeeAction", [](const std::string& remote, boost::optional<bool> addECS) {
-      return std::shared_ptr<DNSAction>(new TeeAction(ComboAddress(remote, 53), addECS ? *addECS : false));
+  luaCtx.writeFunction("TeeAction", [](const std::string& remote, boost::optional<bool> addECS, boost::optional<std::string> local) {
+      return std::shared_ptr<DNSAction>(new TeeAction(ComboAddress(remote, 53), ComboAddress(local ? *local : "0.0.0.0", 0), local ? true : false, addECS ? *addECS : false));
```

Comment:
I agree with Otto, we could pass a `boost::optional<ComboAddress>` to the TeeAction constructor, something like:
```c++
boost::optional<ComboAddress> localAddr{boost::none};
if (local) {
  localAddr = ComboAddress(*local);
}
return std::shared_ptr<DNSAction>(new TeeAction(ComboAddress(remote, 53), localAddr, addECS ? *addECS : false));
```

Reply:
Agree, I will make a commit.

Reply:
Commit in https://github.com/PowerDNS/pdns/pull/11889/commits/8dbb2c9b9da9d4cb1864158f79f59cd2f50efa89